packages/next-drupal/src/types/options.ts

```diff
@@ -6,7 +6,7 @@ export type Locale = string
 
 export type PathPrefix = string
 
-export interface FetchOptions extends RequestInit {
+export interface FetchOptions extends RequestInit, JsonApiWithNextFetchOptions {
```

Comment:
Popping in on this one based on my ongoing review of #787 Specifically: https://github.com/chapter-three/next-drupal/pull/787/files#diff-5b898e58a9c454bff499b9401f4794981861fc3d8870905f57e8082c4e24b6f5

The notes about NextFetchRequestConfig caused me to take a closer look at the types to confirm that this would actually be possible in the state of the codebase that the Docs PR is in. I believe it is possible to pass options like `next: { revalidate: 3600 }` - our `FetchOptions` interface extends Next's `RequestInit` which does have the `next` property that accepts options for `revalidate` and `tags`. That was a mouthful. Here's what the types look like:

Our fetch options:
```
export interface FetchOptions extends RequestInit {
  withAuth?: boolean | NextDrupalAuth
}
```

Next Types:

```
interface RequestInit {
  next?: NextFetchRequestConfig | undefined
}`

interface NextFetchRequestConfig {
  revalidate?: number | false
  tags?: string[]
}
```

Based on reading that code, I'd expect this to already be supported without this `JsonApiWithNextFetchOptions` type. But my guess is that you saw otherwise when actually working with this.

Reply:
Thanks for bring this up @backlineint. Looking at this again, I noticed that in "our" functions (`getResource`, `createResource`, etc) the options parameter is of type `JsonApiOptions` and not `FetchOptions`. `JsonApiOptions` doesn't contain many of the properties that `FetchOptions` has. We should take a closer look at this and probably rework `JsonApiOptions` to include all/some of `FetchOptions`.

Reply:
It's becoming a bit problematic not being able to pass cache options with next-drupal API requests. 

For example:

* We can't specify Next fetch cache options such as `revalidate`. 
* We can't include Drupal cache-tags and use granular `revalidateTag` cache tag invalidation. 
* Next v15, now RC, no longer caches by default so it becomes more important to be able to pass cache options with the fetch requests. 

What needs doing to progress this one?

Reply:
Agreed @davidwhthomas - working on setting some time aside this week to try to help move this one forward.

While I'm here, one thing that either needs to be added here, or handled in a follow up issue is adding the api route for tag invalidation to the basic (app router) starter.